File: 334-increasing-triplet-subsequence/334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        
        a = b = float("inf")
        
        for n in nums:
            if n > b:
                return True 
            if n <= a:
                a = n
            elif n <= b:
                b = n
        
        return False

File: 334-increasing-triplet-subsequence/334-increasing-triplet-subsequence.c


bool increasingTriplet(int* nums, int numsSize){
    int a = INT_MAX, b = INT_MAX;
    
    for(int i = 0; i < numsSize; i++){
        int n = nums[i];
        if(n > b)
            return true;
        if(n <= a)
            a = n;
        else if(n <= b)
            b = n;
    }
    return false;
}